File: 66-plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;

        // Iterate from the least significant digit
        for (int i = n - 1; i >= 0; i--) {
            // Increment the current digit
            digits[i]++;
            // If the digit becomes less than 10 after incrementing, no carry, so return digits
            if (digits[i] < 10) {
                return digits;
            } else {
                // If there's a carry, set the current digit to 0 and continue with the loop
                digits[i] = 0;
            }
        }

        // If we reach here, it means there's a carry in the most significant digit
        // So, we need to create a new array with an additional digit
        int[] result = new int[n + 1];
        result[0] = 1; // Set the most significant digit to 1
        return result;
    }
}

File: 66-plus-one/plus-one.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        n = len(digits)
    
        # Initialize carry to 1 for the addition
        carry = 1
        result = []
        
        # Traverse the digits in reverse order
        for i in range(n - 1, -1, -1):
            total = digits[i] + carry
            
            # Update the carry
            carry = total // 10
            
            # Update the current digit
            result.insert(0, total % 10)
        
        # If there's a remaining carry, add it to the front
        if carry:
            result.insert(0, carry)
        
        return result
